Fix: Ensure consistency in editor tools for navigation buttons and delete options

sarthaknagoshe2002 opened the pull request:
Fixes: #63991
<!-- Thanks for contributing to Gutenberg! Please follow the Gutenberg Contributing Guidelines:
https://github.com/WordPress/gutenberg/blob/trunk/CONTRIBUTING.md -->

## What?
This PR addresses inconsistencies in the editor tools UI, focusing on:

Navigation Button Design: Ensures consistent use of chevrons for menu items like Shadows and Typography/Font Sizes, which navigate to edit screens.
Delete All Options: Adds or aligns "Delete All" functionality for custom settings like Shadows, Custom Font Sizes, and Colors.

## Why?
Inconsistent UI elements create confusion for users and hinder the overall experience. Aligning the design and functionality of editor tools improves usability, accessibility, and coherence across the interface.

## How?
Updated navigation button components to uniformly display chevrons for items that navigate to sub-settings (e.g., Shadows, Typography/Font Sizes).
Implemented consistent "Delete All" options for applicable settings, ensuring uniform behavior across Shadows, Custom Font Sizes, and Colors.
Adjusted CSS and component logic for styling and interactions to reflect these changes.

## Testing Instructions

1. Navigation Buttons

    - Check that Shadows and Typography/Font Sizes now show chevrons consistently.

2. Delete All Options

    - Verify the "Delete All" option exists and works for Shadows, Custom Font Sizes, and Colors.

## Screenshots or screencast <!-- if applicable -->

https://github.com/user-attachments/assets/dfb019df-d454-4ba4-abb1-7d0456af6d85

sarthaknagoshe2002 committed:
fix: ensure consistency in editor tools for navigation buttons and delete options

sarthaknagoshe2002 committed:
Merge branch 'WordPress:trunk' into fix/issue-63991

sarthaknagoshe2002 commented:
@mikachan Could you kindly review this PR?

mikachan commented on packages/edit-site/src/components/global-styles/shadows-panel.js:
```diff
@@ -21,6 +27,10 @@ import Subtitle from './subtitle';
 import ScreenHeader from './header';
 import { getNewIndexFromPresets } from './utils';
+import { IconWithCurrentColor } from './icon-with-current-color';
+const { Menu } = unlock( componentsPrivateApis );
```
I'd suggest moving `Menu` further down to around line 36, after where `useGlobalSetting` is defined. We don't have a lint rule for it, but it's strange to me seeing a `const` in between the imports.

mikachan commented on packages/edit-site/src/components/global-styles/shadows-panel.js:
```diff
@@ -115,6 +152,23 @@ function ShadowList( { label, shadows, category, canCreate, onCreate } ) {
+					>
+						<Menu.Item onClick={ onReset }>
+							<Menu.ItemLabel>
+								{ __( 'Remove all shadows' ) }
```
```suggestion
								{ __( 'Remove all custom shadows' ) }
```

I think we should be more specific that this will remove just the custom shadows.

mikachan commented on packages/edit-site/src/components/global-styles/shadows-panel.js:
```diff
@@ -115,6 +152,23 @@ function ShadowList( { label, shadows, category, canCreate, onCreate } ) {
+							<Button
+								size="small"
+								icon={ moreVertical }
+								label={ __( 'shadow options' ) }
```
```suggestion
								label={ __( 'Shadow options' ) }
```

mikachan commented on packages/edit-site/src/components/global-styles/shadows-panel.js:
```diff
@@ -135,9 +189,14 @@ function ShadowItem( { shadow, category } ) {
+					<IconWithCurrentColor icon={ shadowIcon } />
+					<FlexItem> { shadow.name } </FlexItem>
+				</HStack>
+				<IconWithCurrentColor icon={ chevronRight } />
```
It looks like the chevrons were added in https://github.com/WordPress/gutenberg/pull/67720. We should merge trunk into this PR and resolve any conflicts.

sarthaknagoshe2002 replied:
I believe the approach I used to add the chevrons is better suited to accommodate the changes proposed in this PR. I’d like to continue with the same method, but please let me know if you have suggestions for a better approach.

mikachan replied:
Could you explain the approach you're referring to here? At the moment it looks like this PR is using the same method to add the chevrons:

`<Icon icon={ isRTL() ? chevronLeft : chevronRight } />`

sarthaknagoshe2002 replied:
I was referring to the extra `Hstack`, however now as we have decided to remove it lets ignore that 😇

mikachan reviewed (commented):
Nice work! I can see the shadow options are now more in line with the typography options, with the additional options menu and the confirm modal.

| trunk | Typography | This PR |
| ---- | ---- | ---- |
| ![image](https://github.com/user-attachments/assets/1e061316-e1df-4a22-9e7a-f712db69d5d2) | ![image](https://github.com/user-attachments/assets/f5260e34-b2a4-45f5-8ad7-704aad9f8717) | ![image](https://github.com/user-attachments/assets/5f311bf1-ac55-4f4b-91f0-89eeebab239e) |

I've left a couple of inline comments. I think we should resolve the conflicts with trunk and then I'll give this another review 🙌

sarthaknagoshe2002 committed:
Fix: resolve merge conflicts & address the changes

sarthaknagoshe2002 committed:
Merge branch 'WordPress:trunk' into fix/issue-63991

sarthaknagoshe2002 commented:
@mikachan I’ve resolved the merge conflicts and addressed the feedback.  

I also noticed that a `tsconfig.json` file added yesterday to the latest trunk has some linting issues, which were automatically rectified by the pre-commit git hooks. Unfortunately, I’m unable to undo those changes, as the linting check requires proper formatting before committing.  

Could you please share any insights or suggestions on how to handle this?

sarthaknagoshe2002 commented:
> @mikachan I’ve resolved the merge conflicts and addressed the feedback.
> 
> I also noticed that a `tsconfig.json` file added yesterday to the latest trunk has some linting issues, which were automatically rectified by the pre-commit git hooks. Unfortunately, I’m unable to undo those changes, as the linting check requires proper formatting before committing.
> 
> Could you please share any insights or suggestions on how to handle this?

This is now resolved.

mikachan commented on packages/upload-media/tsconfig.json:
```diff
@@ -8,13 +8,13 @@
-	  { "path": "../i18n" },
-	  { "path": "../private-apis" },
-	  { "path": "../url" }
+		{ "path": "../api-fetch" },
+		{ "path": "../blob" },
+		{ "path": "../compose" },
+		{ "path": "../data" },
+		{ "path": "../element" },
+		{ "path": "../i18n" },
+		{ "path": "../private-apis" },
+		{ "path": "../url" }
```
It looks like this has been resolved on trunk so we should be able to discard the changes to this file.

mikachan commented on packages/edit-site/src/components/global-styles/shadows-panel.js:
```diff
@@ -136,11 +195,12 @@ function ShadowItem( { shadow, category } ) {
-				</FlexItem>
+			<HStack justify="space-between">
+				<HStack justify="flex-start">
+					<IconWithCurrentColor icon={ shadowIcon } />
```
The `shadowIcon` was removed in https://github.com/WordPress/gutenberg/pull/67720 as per [this comment](https://github.com/WordPress/gutenberg/issues/67696#issuecomment-2525550554), so I don't think we should introduce it again here.

If we remove the `shadowIcon`, then I think we can also simplify the markup in line with [the changes](https://github.com/WordPress/gutenberg/pull/67720/files#diff-d76fe0b9848e786a8cc26155bd54635fb31ed7bb5ba193af1e702fe7c1cbeaaeR139) in the above PR (removing the additional `HStack`).

sarthaknagoshe2002 replied:
This is now fixed.

sarthaknagoshe2002 committed:
Merge branch 'WordPress:trunk' into fix/issue-63991

sarthaknagoshe2002 committed:
Fix: remove shadow icon

mikachan commented on packages/edit-site/src/components/global-styles/shadows-panel.js:
```diff
@@ -136,11 +193,9 @@ function ShadowItem( { shadow, category } ) {
 			path={ `/shadows/edit/${ category }/${ shadow.slug }` }
 		>
-			<HStack>
+			<HStack justify="space-between">
```
I don't think `justify="space-between"` is required, as it's the default alignment for this component.

mikachan commented on packages/edit-site/src/components/global-styles/shadows-panel.js:
```diff
@@ -40,8 +50,27 @@ export default function ShadowsPanel() {
+			{ isResetDialogOpen && (
+				<ConfirmResetShadowDialog
+					text={ __(
+						'Are you sure you want to remove all custom Shadows?'
```
```suggestion
						'Are you sure you want to remove all custom shadows?'
```

Small nitpick for consistency, as I don't think we capitalise this anywhere else.

sarthaknagoshe2002 committed:
Fix: remove redandunt styling

sarthaknagoshe2002 committed:
Fix: use lower case in confirmation message

Co-authored-by: Sarah Norris <1645628+mikachan@users.noreply.github.com>

mikachan reviewed (approved):
Thanks for your work here, @sarthaknagoshe2002! I think this is ready to merge when the tests pass 🎉
